tools/ci_build/github/azure-pipelines/templates/linux-build-step-with-cache.yml

```diff
@@ -0,0 +1,50 @@
+parameters:
+- name: WithCache
+  displayName: Build with Cache
+  type: boolean
+  default: false
+
+- name: BuildStep
+  type: stepList
+
+- name: Today
+  type: string
+  default: ""
+
+# It'd be different in different jobs in one workflow.
+- name: AdditionalKey
+  type: string
+  default: ""
+
+- name: ChangeEveryCommit
+  type: boolean
+  default: true
+
+- name: CacheDir
+  type: string
+  default: ""
+
+steps:
+  - script: |
+      mkdir -p ${{ parameters.CacheDir }}
+    displayName: Create Cache Dir
+    condition: eq(${{ parameters.WithCache }}, true)
```

Comment:
Here you can use a template

${{ if eq(parameters.WithCache, true) }}:
  - script: |
      mkdir -p ${{ parameters.CacheDir }}
    displayName: Create Cache Dir

A template is processed when a pipeline run is submitted.     "condition" is only evaluated when this step is ready to run.  template is more static.  If you use a template, this step will not show up in the task list when WithCache is false.

Reply:
thanks!. updated